### ISSUE TITLE
B2C with MSAL.NET sample does not include `.WithRedirectUri()`

### ISSUE THREAD

**Issue by jennyf19** [xamarin/xamarin-forms-samples#555]
The B2C [sample](https://github.com/xamarin/xamarin-forms-samples/blob/master/WebServices/AzureADB2CAuth/ADB2CAuthorization/App.xaml.cs#L19) using MSAL.NET does not include [`.WithRedirectUri()`](https://github.com/Azure-Samples/active-directory-b2c-xamarin-native/blob/master/UserDetailsClient/UserDetailsClient.Core/Features/LogOn/B2CAuthenticationService.cs#L23), so B2C does not know where to callback with the tokens. Customers have lost time trying to troubleshoot and figure this out. Brought to our attention [here](https://github.com/Azure-Samples/active-directory-b2c-xamarin-native/issues/28#issuecomment-555607418)

Will submit a PR to update.

cc: @jmprieur @n8allan

**Comment by profexorgeek** [comment]
Hi @jennyf19 I'm looking into this to update our sample but I'm hoping you can help me understand something. I understand the proposed fix is to add the `WithRedirectUri` method when instantiating the `AuthenticationResult`. But I do not understand why this is necessary or why it changed...

In the Azure Portal, you can specify a custom Redirect URI, for example:
![image](https://user-images.githubusercontent.com/711100/70092427-887aed00-15db-11ea-9d62-97eed22c91f4.png)

If you specify a custom Redirect URI in Azure, why do you also have to specify it in the app? Isn't the Azure ADB2C instance responsible for handing control back to the application with the custom schema?

**Comment by jmprieur** [comment]
@profexorgeek : the redirect URI(s) (note the 's') that you register in Azure are the URIs where the B2C service will agree to send back tokens. You app needs to specify a redirect URI it uses (and there is a specific format in the case of iOS and Android so that your app can be called back even when using a system browser). This redirect URI specified by the app needs to be one of the redirect URIs accepted by Azure.

**Comment by profexorgeek** [comment]
Ah okay. Thank you for the response! I'll get the changes made to our docs/samples.

**Comment by jennyf19** [comment]
@profexorgeek thanks so much!

**Comment by profexorgeek** [comment]
@jennyf19 and @jmprieur just one more comment for posterity. I had to poke at our sample a little more because it _works_ for me without using the `WithRedirectUri` method. It bothered me that it worked for me but not for the user and I had to understand why.

The key detail  here, which @jmprieur said but I missed, is that _if you have multiple redirect URLs in Azure_, it doesn't know which one to use and `WithRedirectUri` provides that information. But if you have a single URI (which is the case in my configuration) it _accidentally works_ because Azure defaults to that redirect. That's the only reason I can think of why the sample would work for me but not the user. 

Obviously the best practice would be to always specify using `WithRedirectUri` but I think it's worth noting this behavior in our doc.
